sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2781,13 +2782,18 @@ class AstBuilder extends SqlBaseParserBaseVisitor[AnyRef] with SQLConfHelper wit
     Option(commentSpec()).map(visitCommentSpec).foreach {
       builder.putString("comment", _)
     }
+    // Add the 'DEFAULT expression' clause in the column definition, if any, to the column metadata.
+    Option(ctx.defaultExpression()).map(visitDefaultExpression).foreach { field =>
```

Comment:
nit: We don't need function visitDefaultExpression. We can define a variable in the grammar file like
```
defaultExpression
    : DEFAULT defaultValue=expression
```
and then we can have 
```
Option(ctx.defaultExpression()).map(_.defaultValue.getText)
```
here

Reply:
Good idea, done.